src/components/container.rs

```diff
@@ -38,11 +44,45 @@ pub fn calendar_view_container(oob: bool) -> Markup {
     }
 }
 
-pub fn courses_container(oob: bool) -> Markup {
+pub fn courses_container(
+    oob: bool,
+    term: Term,
+    courses: &[Course],
+    selected: &SelectedCourses,
+) -> Markup {
+    // TODO: add selected section selection endpoints (i.e PUT /term/:term/calendar/section crn=1)
     html! {
-        div id="courses-container" hx-swap-oob=[if oob {Some("true")} else {None}] class="flex flex-col gap-2 h-full shrink-0 grow basis-1/2 lg:basis-1/5" {
-            div id="courses-card" class="rounded-lg h-full bg-white dark:bg-neutral-800 flex justify-center items-center p-1 dark:text-white" {
-                "selected courses"
+        section class="h-full overflow-y-hidden shrink-0 grow basis-1/2 lg:basis-1/5 bg-white dark:bg-neutral-800 p-2" {
+            h2 class="text-black dark:text-white text-center text-xl font-semibold my-3" {
+                "Selected Courses"
+            }
+
+            div id="courses-container" hx-swap-oob=[if oob {Some("true")} else {None}]
+                class="flex flex-col gap-2 h-full overflow-y-scroll" {
+                @for course in courses {
+                    div id="courses-card" class="rounded-lg flex-col justify-center items-center p-1 dark:text-white border border-black dark:border-white" {
+                        details class="py-3" {
+                            summary class="flex cursor-pointer" {
+                                div class="flex gap-2 justify-between items-center" {
+                                    h3 {
+                                        (&course.subject_code) " " (&course.course_code) " - " (&course.title)
+                                    }
+                                    form {
+                                        input type="hidden" name="subject_code" value=(course.subject_code){}
+                                        input type="hidden" name="course_code" value=(course.course_code){}
+                                        button name="course" value={(course.subject_code) " " (course.course_code)}
```

Comment:
There's actually no request body. HTMX encodes form values as URL parameters on DELETE requests to comply with the HTML spec.
https://github.com/bigskysoftware/htmx/blob/d152a3c75a8a2bd905379de068e6a79a7dddcca0/src/htmx.js#L230-L235

But I think I see what you're getting at: PUT and DELETE to `/term/:term/calendar` both only take a single parameter, `course`, which is a space-delimited string that parses to a `ThinCourse`.

https://github.com/brennanmcmicking/scheduler/blob/7bd800cea01173e3f1eaf9311800cfd96dadfdcf/src/routes/calendar.rs#L18-L29

We can remove the hidden form elements in both `search_result::render` and `container::courses_container` and it'll behave the exact same.

Reply:
:+1: for url param, but I'm not sure we can remove these hidden fields for this, or at least I couldn't come up with a solution that would allow so. 

The raw formdata string is `course_code=xxx&section_code=xxx&crn=1&crn=2...`. All of the checkbox values in the form will be sent, (hence why the html it generates has a biiigggg form), the server just replaces the internal crn seletions with whatever that is sent to the server

I need the course code and section code to search for selected crns in user state cookie.

I'd be happy to re-implement this if you guys can think of a sane solution

Reply:
apparently I was thinking of something else entirely, ignore me

Reply:
I think I see what you might have been going for. In my mind there are two options:

- The user clicks a radio button and triggers form resubmission for just that button group. This acts as a diff, where for example, the lecture section's CRN is replaced with the one submitted. This puts the idea of "section types" into the handler code, as logic that requires database lookups (we need crn -> sequence_code to do this).
- The user clicks a radio button and triggers form resubmission for all of that course's section types. This acts as a full replacement, and all selected radio buttons are submitted to replace all previous state for that course with, simply, all currently selected radio buttons. This makes the submission handler simpler, requiring no additional database lookups (just blindly replace the state for that course with listed CRNs), and moves the grouping logic to the view. In the view we already have seqence_codes (required for grouping radio buttons in UI by i.e. A01, A02, B01, B01). The downside is it provides less validation in the backend that submitted data is "correct" (i.e. someone selects two lecture sections in devtools somehow).

Both of these options are totally fine imo, and I'm curious to hear both of your opinions on this @brennanmcmicking @hn275. Feel free to implement either approach, but I think it's important to document why we might choose either approach, and what goes into that decision, at least in this PR thread to refer back to later.